webapp/src/features/adminEditIntersection/adminEditIntersectionSlice.tsx

```diff
@@ -0,0 +1,234 @@
+      token,
+      query_params: { intersection_id },
+      additional_headers: { 'Content-Type': 'application/json' },
+    })
+
+    switch (data.status) {
+      case 200:
+        dispatch(adminEditIntersectionSlice.actions.updateStates(data.body))
+        return { success: true, message: '', data: data.body }
+      default:
+        return { success: false, message: data.message }
+    }
+  },
+  { condition: (_, { getState }) => selectToken(getState() as RootState) != undefined }
+)
+
+export const editIntersection = createAsyncThunk(
+  'adminEditIntersection/editIntersection',
+  async (json: { orig_intersection_id: string }, { getState, dispatch }) => {
+    const currentState = getState() as RootState
+    const token = selectToken(currentState)
+
+    const data = await apiHelper._patchData({
+      url: EnvironmentVars.adminIntersection,
+      token,
+      query_params: { intersection_id: json.orig_intersection_id },
+      body: JSON.stringify(json),
+    })
+
+    switch (data.status) {
+      case 200:
+        dispatch(updateIntersectionTableData())
+        return { success: true, message: 'Changes were successfully applied!' }
+      default:
+        return { success: false, message: data.message }
+    }
+  },
+  { condition: (_, { getState }) => selectToken(getState() as RootState) != undefined }
+)
+
+export const submitForm = createAsyncThunk(
+  'adminEditIntersection/submitForm',
+  async (data: AdminEditIntersectionFormType, { getState, dispatch }) => {
+    const currentState = getState() as RootState
+    if (checkForm(currentState.adminEditIntersection)) {
+      let json = updateJson(data, currentState.adminEditIntersection)
+      let res = await dispatch(editIntersection(json))
+      if ((res.payload as any).success) {
+        return { submitAttempt: false, success: true, message: 'Intersection Updated Successfully' }
+      } else {
+        return { submitAttempt: false, success: false, message: (res.payload as any).message }
+      }
+    } else {
+      return { submitAttempt: true, success: false, message: 'Please fill out all required fields' }
+    }
+  }
+)
+
+export const adminEditIntersectionSlice = createSlice({
+  name: 'adminEditIntersection',
+  initialState: {
+    loading: false,
+    value: initialState,
+  },
+  reducers: {
+    setSelectedOrganizations: (state, action) => {
+      state.value.selectedOrganizations = action.payload
+    },
+    setSelectedRsus: (state, action) => {
+      state.value.selectedRsus = action.payload
+    },
+    updateStates: (state, action: { payload: adminEditIntersectionData }) => {
+      const apiData = action.payload
+
+      const allowedSelections = apiData.allowed_selections
+      state.value.organizations = allowedSelections.organizations.map((val) => {
+        return { name: val }
+      })
+      state.value.rsus = allowedSelections.rsus.map((val) => {
+        return { name: val?.replace('/32', '') }
```

Comment:
Are we just updating the rsu ip address to be more human readable here by removing the CIDR subnet mask? Can we add some comments into these to say why we're doing some of this so we don't forget later?

Reply:
Added comments, these are both for human readability and api compatibility (for the editIntersection API)